package/kedro_viz/integrations/kedro/lite_parser.py

```diff
@@ -0,0 +1,199 @@
+            )
+            if (
+                not self._is_module_importable(full_module_name)
+                and full_module_name not in mocked_modules
+            ):
+                mocked_modules[full_module_name] = MagicMock()
+
+    def _populate_mocked_modules(
+        self,
+        parsed_content_ast_node: ast.Module,
+        mocked_modules: Dict[str, MagicMock],
+    ) -> None:
+        """Populate mocked_modules with missing external dependencies
+
+        Args:
+            parsed_content_ast_node (ast.Module): The AST node to
+                    extract import statements
+            mocked_modules (Dict[str, MagicMock]): A dictionary of mocked imports
+        """
+        for node in ast.walk(parsed_content_ast_node):
+            # Handling dependencies that starts with "import "
+            # Example: import logging
+            # Corresponding AST node will be:
+            # Import(names=[alias(name='logging')])
+            if isinstance(node, ast.Import):
+                for alias in node.names:
+                    module_name = alias.name
+                    self._create_mock_imports(module_name, mocked_modules)
+
+            # Handling dependencies that starts with "from "
+            # Example: from typing import Dict, Union
+            # Corresponding AST node will be:
+            # ImportFrom(module='typing', names=[alias(name='Dict'),
+            #            alias(name='Union')],
+            #            level=0)
+            elif isinstance(node, ast.ImportFrom):
+                module_name = node.module if node.module else ""
+                level = node.level
+
+                # Ignore relative imports like "from . import a"
+                if not module_name:
+                    continue
+
+                # Ignore relative imports within the package
+                # Examples:
+                # "from demo_project.pipelines.reporting import test",
+                # "from ..nodes import func_test"
+                if (self._package_name and self._package_name in module_name) or (
+                    # dev or standalone mode
+                    not self._package_name
+                    and self._is_relative_import(module_name)
+                ):
+                    continue
+
+                # absolute modules in the env
+                # Examples:
+                # from typing import Dict, Union
+                # from sklearn.linear_model import LinearRegression
+                if level == 0:
+                    self._create_mock_imports(module_name, mocked_modules)
+
+    def get_mocked_modules(self) -> Dict[str, MagicMock]:
+        """Returns mocked modules for all the dependency errors
+        as a dictionary for each file in your Kedro project
+        """
+        mocked_modules: Dict[str, MagicMock] = {}
+
+        for file_path in self._project_file_paths:
+            with open(file_path, "r", encoding="utf-8") as file:
```

Comment:
Nit: Can it be less couple with viz/project structure. Let say I want to mock an arbitary file, I should be able to reuse the same class.

i.e maybe it could be `def get_mocked_modules(self, paths)` instead? This should make testing much easier as you can test on a single file.